Terminal.Gui/Windows/FileDialog2.cs

```diff
@@ -0,0 +1,2118 @@
+				var tile = this.splitContainer.Tiles.ElementAt (0);
+
+				var newState = !tile.ContentView.Visible;
+				tile.ContentView.Visible = newState;
+				this.btnToggleSplitterCollapse.Text = newState ? "<<" : ">>";
+			};
+
+
+			tbFind = new CaptionedTextField {
+				X = Pos.Right (this.btnToggleSplitterCollapse) + 1,
+				Caption = "Enter Search",
+				Width = 16,
+				Y = Pos.AnchorEnd (1),
+			};
+			spinnerLabel = new SpinnerLabel () {
+				X = Pos.Right (tbFind) + 1,
+				Y = Pos.AnchorEnd (1),
+				Width = 1,
+				Height = 1,
+				Visible = false,
+			};
+
+			tbFind.TextChanged += (o) => RestartSearch ();
+
+			lblFeedback = new Label {
+				Y = Pos.AnchorEnd (1),
+				X = Pos.Right (spinnerLabel) + 1,
+				ColorScheme = new ColorScheme {
+					Normal = new Attribute (Color.Red, this.ColorScheme.Normal.Background)
+				}
+			};
+
+			this.tableView.Style.ShowHorizontalHeaderOverline = false;
+			this.tableView.Style.ShowVerticalCellLines = false;
+			this.tableView.Style.ShowVerticalHeaderLines = false;
+			this.tableView.Style.AlwaysShowHeaders = true;
+
+
+			this.SetupColorSchemes ();
+
+			this.SetupTableColumns ();
+
+			this.sorter = new FileDialogSorter (this, this.tableView);
+			this.history = new FileDialogHistory (this);
+
+			this.tableView.Table = this.dtFiles;
+
+			this.tbPath.TextChanged += (s) => this.PathChanged ();
+
+			this.tableView.CellActivated += this.CellActivate;
+			this.tableView.KeyUp += (k) => k.Handled = this.TableView_KeyUp (k.KeyEvent);
+			this.tableView.SelectedCellChanged += this.TableView_SelectedCellChanged;
+			this.tableView.ColorScheme = ColorSchemeDefault;
+
+			this.tableView.AddKeyBinding (Key.Home, Command.TopHome);
+			this.tableView.AddKeyBinding (Key.End, Command.BottomEnd);
+			this.tableView.AddKeyBinding (Key.Home | Key.ShiftMask, Command.TopHomeExtend);
+			this.tableView.AddKeyBinding (Key.End | Key.ShiftMask, Command.BottomEndExtend);
+
+
+			this.treeView.ColorScheme = ColorSchemeDefault;
+			this.treeView.KeyDown += (k) => {
+
+
+				var selected = treeView.SelectedObject;
+				if (selected != null) {
+					if (!treeView.CanExpand (selected) || treeView.IsExpanded (selected)) {
+						this.NavigateIf (k, Key.CursorRight, this.tableView);
+					} else
+					if (treeView.GetObjectRow (selected) == 0) {
+						this.NavigateIf (k, Key.CursorUp, this.tbPath);
+					}
+				}
+
+				if (k.Handled) {
+					return;
+				}
+
+				k.Handled = this.TreeView_KeyDown (k.KeyEvent);
+
+			};
+
+			this.AllowsMultipleSelection = false;
+
+			this.UpdateNavigationVisibility ();
+
+			// Determines tab order
+			this.Add (this.btnToggleSplitterCollapse);
+			this.Add (this.tbFind);
+			this.Add (this.spinnerLabel);
+			this.Add (lblFeedback);
+			this.Add (this.btnOk);
+			this.Add (this.btnCancel);
+			this.Add (this.lblUp);
+			this.Add (this.lblBack);
+			this.Add (this.lblForward);
+			this.Add (lblPath);
+			this.Add (this.tbPath);
+			this.Add (this.splitContainer);
+		}
+		/// <inheritdoc/>
```

Comment:
Please be consistent in your whitespace (blank lines). E.g. here you have none, a little further down you have 2.